### ISSUE TITLE
[feature]: equivalent of fzf --with-nth

### ISSUE THREAD

**Issue by justchokingaround** [charmbracelet/gum#16]
let's say, i have for instance the following file.txt:
```
useful dont_show number one
also_useful idk number two
```
i would like to filter out some information, and only display a portion of it to the user. with fzf, i could do the following 
`fzf --with-nth 3.. < file.txt` 

<img width="158" alt="image" src="https://user-images.githubusercontent.com/44473782/181610253-8d45797c-07cd-470d-96f2-add43980debf.png">
and after i select one of the entries, i get back the whole line with other useful information, which i would not like to be displayed, but would like to use

<img width="277" alt="image" src="https://user-images.githubusercontent.com/44473782/181610500-bbd0fbd8-f93d-4ca1-8613-22318a765f34.png">

would it be possible for this feature to be implemented, and ideally would it be possible to also integrate regex matching into it, instead of using awk/cut delimiters, like fzf is currently doing? hopefully i was clear enough, sorry if some parts might've sounded confusing

**Comment by maaslalani** [comment]
Hey @justchokingaround, the following is equivalent to the `fzf --with-nth 3..` (`cut -d' ' -f3-`) and I believe it's more flexible and more robust than absorbing the complexity into `gum`.

```
cut -d' ' -f3- < file.txt | gum filter
```

Is there any reason you would prefer `gum` to have this functionality than using other `unix` tools to do this?

**Comment by justchokingaround** [comment]
well the whole point of that functionality is to use it for scripting. i want to display a selected portion but get back the whole line

**Comment by maaslalani** [comment]
Yeah, you're totally right, I understand the problem now. It's definitely doable with `grep` after the fact but definitely not nice or user-friendly and would break if there are duplicates most likely.

I'll look at a way to make `gum filter` more flexible 👍

**Comment by justchokingaround** [comment]
ty very much (pls pls integrate regex matching if possible) 🙏
